src/mpl/include/mpl_thread_priv.h

```diff
@@ -7,7 +7,21 @@
 #ifndef MPL_THREAD_PRIV_H_INCLUDED
 #define MPL_THREAD_PRIV_H_INCLUDED
 
-#if MPL_THREAD_PACKAGE_NAME != MPL_THREAD_PACKAGE_NONE && !defined(MPL_TLS)
+extern int MPL_is_threaded;
+void MPL_set_threaded(int isThreaded);
+
+#if MPL_THREAD_PACKAGE_NAME < MPL_THREAD_PACKAGE_YES
+/* No thread package, no need for tls */
+#undef MPL_TLS
+#define MPL_TLS
+
+#elif MPL_THREAD_PACKAGE_NAME > MPL_THREAD_PACKAGE_NON_NATIVE
+/* compiler tls won't work in this case */
+#undef MPL_TLS
+
+#endif
+
```

Comment:
Is it possible to define `MPL_TLS` etc in `mpl_thread_xxx.h` instead? I don't think this comparison-based detection is clean.

Reply:
My reasoning is to let `configure` do feature detection and only feature detection. So it detects whether compiler support TLS keyword and defines `MPL_TLS`. That is why that is defined in `mplconfig.h`. 

It is a hack to simplify logic. I agree it is less desirable. How about I create a new macros, e.g. `MPL_TLS_DIRECT` instead of hacking `MPL_TLS`?

Reply:
https://github.com/pmodels/mpich/blob/master/src/mpl/include/mpl_thread.h includes `mpl_thread_priv.h` after `mpl_thread_xxx.h` (e.g., `mpl_thread_posix.h` and `mpl_thread_argobots.h`). Can't we put this logic in these `mpl_thread_xxx.h` headers? If so, we can remove comparison against `MPL_THREAD_PACKAGE_NON_NATIVE` etc.

Changing the name (`MPL_TLS_DIRECT` or `MPL_COMPILER_TLS_SUPPORT`) looks good to me.